src/spikeinterface/core/waveform_extractor.py

```diff
@@ -959,6 +961,19 @@ def get_waveforms(self, unit_id, with_index=False, cache=False, lazy=True, spars
             assert not self.is_sparse(), "Waveforms are alreayd sparse! Cannot apply an additional sparsity."
             wfs = wfs[:, :, sparsity.mask[self.sorting.id_to_index(unit_id)]]
 
+        if force_dense:
+            num_channels = self.get_num_channels()
+            dense_wfs = np.zeros((wfs.shape[0], wfs.shape[1], num_channels), dtype=np.float32)
+            unit_ind = self.sorting.id_to_index(unit_id)
+            if sparsity is not None:
+                unit_sparsity = sparsity.mask[unit_ind]
+                dense_wfs[:, :, unit_sparsity] = wfs
+                wfs = dense_wfs
```

Comment:
I think we should not allow the case sparsity is None and force_dense=True.
This is non sens.
I would handle only internal sparse.
So we could have a checker `assert self.is_sparse()` when `force_dense=True`
no ?

Reply:
In fact, I think it could make sense. Someone may want to impose some sparsity and still, get the waveforms in a dense format

Reply:
I agree with Pierre